scripts/paternity-test.py

```diff
@@ -0,0 +1,173 @@
+            df.at[index_label,'ID'] = child_baseID
+            #output_list.update({baseID(row_series['filename'], child_extension) : df.at[index_label,'FileType']})
+        else:
+            pass
+
+logging.info("Matching children and parents...")
+i = 0 #only used for logging
+# Iterate yet again to match children with their parents
+for child_extension in unique_children: #loop once per child extension
+    i=i+1
+    logging.info("\t* Dealing with %s (child %d out of %d)" % (child_extension, i, len(unique_children)))
+    for index_label, row_series in df.iterrows():
+        if(df.at[index_label,'FileType'] == PARENT_FILETYPE):
+            # Find this parent's child
+            # Child might be above parent so we can't just start from index of the parent
+            for index_label_inception, row_series_inception in df.iterrows():
+                #print("Outer iter %d, inner iter %d parent %s checking if %s is its child" % (index_label, index_label_inception, df.at[index_label,'filename'], df.at[index_label_inception,'filename']))
+                if index_label != index_label_inception: #needed to preventing it find itself
+                    if(df.at[index_label,'ID'] == df.at[index_label_inception,'ID']):
+                        # Child found!
+                        #print("    Found "+df.at[index_label_inception,'filename']+" to be child")
+                        df.at[index_label,child_extension] = df.at[index_label_inception,'filename']
+                        df.at[index_label,child_extension+"_location"] = df.at[index_label_inception,'location']
+                        break
+            if(df.at[index_label,child_extension] == ""):
+                logging.warning("Could not find child of type %s for parent %s" % (child_extension, df.at[index_label, 'FileType']))
+print(df)
+logging.info("Cleaning up dataframe...")           
+# Iterate one more time to delete child rows
+# Because children could appear above their parents, deleting during the above iteration could mess things up
+df.drop(columns=['ID'], inplace=True)
+df.rename(columns = {'FileType':'parent_file_ext'}, inplace = True)
+for index_label, row_series in df.iterrows():
+        if(df.at[index_label,'parent_file_ext'] != PARENT_FILETYPE):
+            df.drop([index_label], inplace=True)
+df.reset_index(inplace=True, drop=True)
+logging.info("Finished")
+
+# Generate TSV file from Dataframe
+
+#!rm final.tsv
+# ^ Uncomment above line if you will be running this block more than once
+
+df.to_csv("dataframe.tsv", sep='\t')
+# Format resulting TSV file to play nicely with Terra 
+with open('dataframe.tsv', "r+") as file1:
+    header = file1.readline()
+    everything_else = file1.readlines()
+    file1.close()
+full_header="entity:"+TABLE_NAME+"_id"+header
+with open('final.tsv', "a") as file2:
+    file2.write(full_header)
+    for string in everything_else:
+        file2.write(string)
+    file2.close()
+
+# Clean up
+!rm dataframe.tsv
```

Comment:
You can use `os.remove("dataframe.tsv")`

Reply:
For future reference, is there any advantage to this or is this just to keep things pythonic (ie avoid bash)?

Reply:
Just to keep things pythonic.